KsApi/px/PerimeterXClient.swift

```diff
@@ -13,59 +17,39 @@ public class PerimeterXClient: NSObject, PerimeterXClientType {
       .domain: "www.perimeterx.com", // Change according to the domain the webview will use
       .path: "/",
       .name: "_pxmvid",
-      .value: self.manager.getVid() as Any,
+      .value: self.vid ?? "",
       .expires: self.dateType.init(timeIntervalSinceNow: 3_600).date
     ])
   }()
 
   public init(
-    manager: PerimeterXManagerType = PXManager.sharedInstance(),
     dateType: ApiDateProtocol.Type = Date.self
   ) {
-    self.manager = manager
     self.dateType = dateType
 
     super.init()
-
-    /// When this isn't a mock we'll set the delegate and have debug logging in the console.
-    (self.manager as? PXManager)?.delegate = self
-  }
-
-  public func handleError(response: HTTPURLResponse, and data: Data) -> Bool {
-    /// We have a `403` statusCode.
-    guard response.statusCode == 403 else { return false }
-
-    guard
-      /// We have `JSON`.
-      let jsonData = try? JSONSerialization.jsonObject(with: data, options: []) as? [String: Any],
-      /// We have a `PXBlockResponse`.
-      let response: PerimeterXBlockResponseType = self.manager.checkError(jsonData),
-      /// The response's `PXBlockType` is `Block` or `Captcha`.
-      [.Block, .Captcha].contains(response.type)
-    else { return false }
-
-    DispatchQueue.main.async {
-      NotificationCenter.default.post(
-        name: Notification.Name.ksr_perimeterXCaptcha,
-        object: nil,
-        userInfo: ["response": response]
-      )
-    }
-
-    return true
   }
 
   public func headers() -> [String: String] {
-    return (self.manager.httpHeaders() as? [String: String]) ?? [:]
+    return PerimeterX.headersForURLRequest() ?? [:]
   }
-}
 
-extension PerimeterXClient: PXManagerDelegate {
-  public func managerReady(_ httpHeaders: [AnyHashable: Any]!) {
-    print("❎ Perimeter X headers ready: \(String(describing: httpHeaders))")
+  public func start(policyDomains: Set<String>) {
+    self.policy.set(domains: policyDomains, forAppId: Secrets.PerimeterX.appId)
+
+    try? PerimeterX.start(appId: Secrets.PerimeterX.appId, delegate: nil, policy: self.policy)
   }
 
-  public func newHeaders(_ httpHeaders: [AnyHashable: Any]!) {
-    print("❎ Perimeter X headers were refreshed: \(String(describing: httpHeaders))")
+  public func handleResponse(data: Data, response: URLResponse) -> Bool {
+    PerimeterX.handleResponse(response: response, data: data) { result in
+      switch result {
+      case .cancelled:
+        print("cancelled")
+      case .solved:
+        print("solved")
```

Comment:
This doesn't feel submit-ready. Does anything need to happen here? Or, if we just want to log something here, can we do it with more complete messaging?

Reply:
Ah good call out - yes there should not be print statements here. Will remove in my next commit.

Reply:
left them in - previous implementation had some [logs](https://github.com/kickstarter/ios-oss/pull/1839/files#diff-e71c4072e0340759413ae73328b199512c4a430f7ff0ff72e6a863f220296c16L38) that were useful for testing, but removed the force crash `fatalError`.

Reply:
Looks good now, thank you!